flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableInputFormat.java

```diff
@@ -0,0 +1,327 @@
+	private String[] partitionColNames;
+	private List<HiveTablePartition> partitions;
+	private transient Deserializer deserializer;
+	private transient List<? extends StructField> fieldRefs;
+	private transient StructObjectInspector oi;
+	private transient InputFormat mapredInputFormat;
+	private transient HiveTablePartition hiveTablePartition;
+
+	public HiveTableInputFormat(
+			JobConf jobConf,
+			Boolean isPartitioned,
+			String[] partitionColNames,
+			List<HiveTablePartition> partitions,
+			BaseRowTypeInfo baseRowTypeInfo) {
+		super(jobConf.getCredentials());
+		this.baseRowTypeInfo = baseRowTypeInfo;
+		this.jobConf = new JobConf(jobConf);
+		this.jobConf = jobConf;
+		this.isPartitioned = isPartitioned;
+		this.partitionColNames = partitionColNames;
+		this.partitions = partitions;
+	}
+
+	@Override
+	public void open(HiveTableInputSplit split) throws IOException {
+		this.hiveTablePartition = split.getHiveTablePartition();
+		StorageDescriptor sd = hiveTablePartition.getStorageDescriptor();
+		jobConf.set(INPUT_DIR, sd.getLocation());
+		try {
+			this.mapredInputFormat = (InputFormat)
+				Class.forName(sd.getInputFormat(), true, Thread.currentThread().getContextClassLoader()).newInstance();
+		} catch (Exception e) {
+			throw new RuntimeException("Unable to instantiate the hadoop input format", e);
+		}
+		ReflectionUtils.setConf(mapredInputFormat, jobConf);
+		if (this.mapredInputFormat instanceof Configurable) {
+			((Configurable) this.mapredInputFormat).setConf(this.jobConf);
+		} else if (this.mapredInputFormat instanceof JobConfigurable) {
+			((JobConfigurable) this.mapredInputFormat).configure(this.jobConf);
+		}
+		this.recordReader = this.mapredInputFormat.getRecordReader(split.getHadoopInputSplit(),
+			jobConf, new HadoopDummyReporter());
+		if (this.recordReader instanceof Configurable) {
+			((Configurable) this.recordReader).setConf(jobConf);
+		}
+		key = this.recordReader.createKey();
+		value = this.recordReader.createValue();
+		this.fetched = false;
+		try {
+			deserializer = (Deserializer) Class.forName(sd.getSerdeInfo().getSerializationLib()).newInstance();
+			Configuration conf = new Configuration();
+			Properties properties = HiveTableUtil.createPropertiesFromStorageDescriptor(sd);
+			SerDeUtils.initializeSerDe(deserializer, conf, properties, null);
+			// Get the row structure
+			oi = (StructObjectInspector) deserializer.getObjectInspector();
+			fieldRefs = oi.getAllStructFieldRefs();
+		} catch (Exception e) {
+			logger.error("Error happens when deserialize from storage file.");
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override
+	public HiveTableInputSplit[] createInputSplits(int minNumSplits)
+			throws IOException {
+		List<HiveTableInputSplit> hiSplit = new ArrayList<>();
+		int splitNum = 0;
+		for (HiveTablePartition partition : partitions) {
+			StorageDescriptor sd = partition.getStorageDescriptor();
+			InputFormat format;
+			try {
+				format = (InputFormat)
```

Comment:
is this dup of `mapredInputFormat` initialized in `open()`?

Reply:
Nop. This method is called in the jobMaster to calc input splits not in every instance.